### DISCUSSION TITLE
stackSentinel.py and stackStripMap.py - cannot find dem (vrt) in read mode

### DISCUSSION THREAD

**Discussion by Carolinah23** [isce-framework/isce2#595] [Q&A]
Hello, 

I encounter a problem using stackSentinel.py and stackStripMap.py with my own DEM. I used this DEM with stripmapApp.py and topsApp.py to do individual interferograms, so I believe there are no problems with the DEM, although these tools only work if the DEM is in the process directory otherwise neither can find the file even with the absolute path.

Now, when I use stackSentinel.py or stackStripMap.py it displays an error when running the first run file (run_01_unpack_topo_reference and run_01_reference respectively). This error was displayed before by topsApp.py when the DEM was outside the process folder. I have tried to use the DEM file in the process folder and outside and neither option has worked. 

Error for stackSentinel.py
![image](https://user-images.githubusercontent.com/54252458/193155029-144aba70-698a-407d-8edf-ef795ca86784.png)

Error for stackStripMap.py
![image](https://user-images.githubusercontent.com/54252458/193154036-55abdb96-817b-4755-9f38-0f46a4f3acd0.png)

I change the path variable each time that I tried to use each of the tools. The vrt file is also in the folder.

Thanks for any help!

This is the configuration file for unpack_topo:
`[Common]
##########################
##########################
[Function-1]
Sentinel1_TOPS : 
dirname : /media/carolina/CarolOffice/01BatonRouge/PSI/Sentinel1/StackProcess/Data/S1A_IW_SLC__1SDV_20150322T000131_20150322T000200_005137_006794_9357.zip
swaths : 1 2 3
orbitdir : /media/carolina/CarolOffice/orbits/Sentinel/Orbits
outdir : /media/carolina/CarolOffice/01BatonRouge/PSI/Sentinel1/StackProcess/reference
auxdir : /media/carolina/CarolOffice/orbits/Sentinel/AUX
bbox : 30.34 30.51 -91.18 -90.97
pol : vv
##########################
##########################
#Call topo to produce reference geometry files
[Function-2]
topo : 
reference : /media/carolina/CarolOffice/01BatonRouge/PSI/Sentinel1/StackProcess/reference
dem : /media/carolina/CarolOffice/01BatonRouge/PSI/Sentinel1/StackProcess/DEM2018_1
geom_referenceDir : /media/carolina/CarolOffice/01BatonRouge/PSI/Sentinel1/StackProcess/geom_reference
numProcess : 1
##########################`

**Comment by AP2607** [comment]
Hi, i encounter the same problem. Did you manage to solve it?

**Reply by Edda1824** [reply]
Same problem with me when running the first "run_files" (run_01_unpack_topo_reference). Any solution yet?

**Reply by geniusinaction** [reply]
If you moved the DEM files since you downloaded them, you might need to run `fixImageXml.py` to update the location in the XML file. I had a similar problem and that fixed it for me.

**Comment by maipleasurre** [comment]
Sorry I'm a bit late sharing this. To resolve this, we must first transcode dem.tif into a .dem file, followed by the creation of the dem.vrt.

These are the command I use to convert dem file:
*disclaimer I use this command on WSL 

1. goes into DEM folder, example: cd mnt/d/YourFolder/DEM
2. gdal_translate -of ISCE -ot Float32 output_hh.tif output_hh.dem
3. gdal_translate -of VRT output_hh.dem output_hh.dem.vrt
4. sed -i 's|<value>output_hh.dem</value>|<value>/mnt/d/Modul_SBAS/DEM/output_hh.dem</value>|' output_hh.dem.xml
5. gdal_translate -of VRT output_hh.dem output_hh.dem.vrt

and after that you can process the stackSentinel with the converted DEM
stackSentinel.py \
-s SLC \
-o ORBITS \
-a AUX \
-d DEM/**output_hh.dem** \
-b "boundingbox" \
-m referencedate\
-w interferogram \

I hope this helps anyone that need this. 
Thank you, have a nice day :)
